doc/Iothub_sdk_options.md

```diff
@@ -0,0 +1,69 @@
+- [IoThub Client Options](#IotHub_option)
+- [Transport Options](#transport_option)
+
+<a name="set_option"></a>
+
+## Setting an Option
+
+Setting and option in the c-sdk is dependant on which api set you are using.  If using the convience layer you would set the options in the api as follows:
+
+```c
+IOTHUB_CLIENT_RESULT IoTHubClient_SetOption(IOTHUB_CLIENT_HANDLE iotHubClientHandle, const char* optionName, const void* value)
+
+IOTHUB_CLIENT_RESULT IoTHubClient_LL_SetOption(IOTHUB_CLIENT_LL_HANDLE iotHubClientHandle, const char* optionName, const void* value)
+```
+
+An example of using a set option:
+
+```c
+bool trace_on = true;
+IoTHubClient_SetOption(sdk_handle, OPTION_LOG_TRACE, &trace_on);
+```
+
+## Available Option
+
+<a name="IotHub_option"></a>
+
+### IoThub_Client
+
+| Option Name        | Option Define              | Value Type         | Description                  |
+|--------------------|----------------------------|--------------------|-------------------------------|
+| `"messageTimeout"` | OPTION_MESSAGE_TIMEOUT     | tickcounter_ms_t*  | Timeout used for message      |
+| `"product_info"`   | OPTION_PRODUCT_INFO        | const char*        | Product information sent to the IoThub service on connection
+| `"TrustedCerts"`   | OPTION_TRUSTED_CERT        | const char*        | Azure Server certificate used to validate TLS connection to iothub|
+
+<a name="transport_option"></a>
+
+### Standard Transport
+
+| Option Name            | Option Define             | Value Type         | Description                  |
+|------------------------|---------------------------|--------------------|-------------------------------|
+| `"logtrace"`           | OPTION_LOG_TRACE          | bool* value         | Turn on and off log trace       |
+| `"sas_token_lifetime"` | OPTION_SAS_TOKEN_LIFETIME | `size_t`* value      | Length of time in seconds used for lifetime of sas token. |
+| `"x509certificate"`    | OPTION_X509_CERT          | const char*        | Sets the x509 certificate used for connection authentication
+| `"x509privatekey"`     | OPTION_X509_PRIVATE_KEY   | const char*        | Sets the private key for the x509 certificate
+| `"x509EccCertificate"` | OPTION_X509_ECC_CERT      | const char*        | Sets the ECC x509 certificate used for connection authentication
+| `"x509EccAliasKey"`    | OPTION_X509_ECC_KEY       | const char*        | Sets the private key for the ECC x509 certificate
+| `"proxy_data"`         | OPTION_HTTP_PROXY         | [HTTP_PROXY_OPTIONS*][http-proxy-object]| Http proxy data object used for proxy connection to IoTHub
+
+### MQTT Transport
+
+| Option Name            | Option Define             | Value Type         | Description                  |
+|------------------------|---------------------------|--------------------|-------------------------------|
+| `"keepalive"`          | OPTION_KEEP_ALIVE         | int*               | Length of time to send keep Alives to service
```

Comment:
What is keep alive?  Are alives sent as d2c message?

Reply:
Keep Alive are for MQTT pings

---
In reply to: [140866080](https://github.com/Azure/azure-iot-sdk-c/pull/241#discussion_r140866080) [](ancestors = 140866080)